nested/elastic.json

```diff
@@ -0,0 +1,368 @@
+                    "uri": "[concat(parameters('moodleCommon').baseTemplateUrl,'elasticconfig.json')]"
+                }
+            },
+            "type": "Microsoft.Resources/deployments"
+        },
+        {
+            "condition": "[parameters('moodleCommon').azureBackupSwitch]",
+            "apiVersion": "2015-01-01",
+            "dependsOn": [
+                "[concat('Microsoft.Compute/virtualMachines/',parameters('moodleCommon').elasticVmName1)]"
+            ],
+            "name": "[concat(parameters('moodleCommon').elasticVmName1,'-Backup')]",
+            "properties": {
+                "mode": "Incremental",
+                "parameters": {
+                    "moodleCommon": {
+                        "value": "[parameters('moodleCommon')]"
+                    },
+                    "vmName": {
+                        "value": "[parameters('moodleCommon').elasticVmName1]"
+                    }
+                },
+                "templateLink": {
+                    "uri": "[concat(parameters('moodleCommon').baseTemplateUrl,'recoveryservicesEnlist.json')]"
+                }
+            },
+            "type": "Microsoft.Resources/deployments"
+        },
+        {
+            "apiVersion": "2015-06-15",
+            "location": "[resourceGroup().location]",
+            "name": "[parameters('moodleCommon').elasticNicName2]",
+            "properties": {
+                "ipConfigurations": [
+                    {
+                        "name": "ipcfg-elastic2",
+                        "properties": {
+                            "privateIPAllocationMethod": "Static",
+                            "privateIPAddress": "[parameters('moodleCommon').elasticVm2IP]",
+                            "subnet": {
+                                "id": "[variables('subnetElasticRef')]"
+                            }
+                        }
+                    }
+                ]
+            },
+            "tags": {
+                "displayName": "Elastic NIC 2"
+            },
+            "type": "Microsoft.Network/networkInterfaces"
+        },
+        {
+            "apiVersion": "[parameters('moodleCommon').computeApi]",
+            "dependsOn": [
+                "[concat('Microsoft.Network/networkInterfaces/', parameters('moodleCommon').elasticNicName2)]"
+            ],
+            "location": "[resourceGroup().location]",
+            "name": "[parameters('moodleCommon').elasticVmName2]",
+            "properties": {
+                "hardwareProfile": {
+                    "vmSize": "[parameters('moodleCommon').elasticVmSku]"
+                },
+                "networkProfile": {
+                    "networkInterfaces": [
+                        {
+                            "id": "[variables('nicRef2')]"
+                        }
+                    ]
+                },
+                "osProfile": {
+                    "adminPassword": "[parameters('moodleCommon').sshPassword]",
```

Comment:
why are we mixing password and ssh auth throughout the app?

Reply:
Good catch and I can only guess this might be an oversight from the original creator. Unified to ssh pubkey auth.